src/IO/H5/VolumeData.cpp

```diff
@@ -350,6 +660,124 @@ void VolumeData::write_volume_data(
   }
 }
 
+// Write new connectivity connections given a std::vector of observation ids
+template <size_t SpatialDim>
+void VolumeData::extend_connectivity_data(
+    const std::vector<size_t>& observation_ids) {
+  for (const size_t& obs_id : observation_ids) {
+    auto grid_names = get_grid_names(obs_id);
+    auto extents = get_extents(obs_id);
+    auto bases = get_bases(obs_id);
+    auto quadratures = get_quadratures(obs_id);
+
+    auto [number_of_blocks, block_number_for_each_element,
+          sorted_element_indices] = compute_and_organize_block_info(grid_names);
+
+    auto sorted_grid_names = sort_by_block(sorted_element_indices, grid_names);
+    auto sorted_extents = sort_by_block(sorted_element_indices, extents);
+
+    size_t total_expected_connectivity = 0;
+    std::vector<int> expected_grid_points_per_block;
+    expected_grid_points_per_block.reserve(number_of_blocks);
+    std::vector<std::array<int, SpatialDim>> h_ref_per_block;
+    h_ref_per_block.reserve(number_of_blocks);
+
+    // Loop over blocks
+    for (size_t j = 0; j < number_of_blocks; ++j) {
+      auto [expected_connectivity_length, expected_number_of_grid_points,
+            h_ref_array] =
+          compute_block_level_properties<SpatialDim>(sorted_grid_names[j],
+                                                     sorted_extents[j]);
+      total_expected_connectivity += expected_connectivity_length;
+      expected_grid_points_per_block.push_back(expected_number_of_grid_points);
+      h_ref_per_block.push_back(h_ref_array);
+    }
+
+    // Instantiates the unordered_map
```

Comment:
Used for what? And `instantiate` typically refers to types and templates. I would phrase this as `// Create the unordered_map used for ...` Same in other locations